File: iteration2/src/main/Student.java
import org.json.JSONException;
import org.json.JSONObject;
import org.json.simple.parser.ParseException;

import com.google.gson.Gson;

import org.json.*;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Random;

public class Student extends Person {

    private int studentNumber;
    private int year;
    private int semester;
    private ArrayList<Student> studentArrayList = new ArrayList<Student>();

    public Student() {
        super();
    }

    public Student(String name, String department,  int year, int studentNumber, int semester, String rank) {
        super(name,department,rank);
        this.studentNumber = studentNumber;
        this.year = year;
        this.semester = semester;
    }

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {

        this.semester = generateStudentSemester(semester);
    }

    public int getStudentNumber() {
        return studentNumber;
    }

    public void setStudentNumber(int studentNumber) {
        this.studentNumber = studentNumber;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public ArrayList<Student> getStudentArrayList() {
        return studentArrayList;
    }

    public void setStudentArrayList(ArrayList<Student> studentArrayList) {
        this.studentArrayList = studentArrayList;
    }

File: iteration3/src/Student.py

import json
import random

from OOPProject.StudentProcess import StudentProcess
from Person import Person
from configparser import ConfigParser


class Student(Person):

    def __init__(self, year = None, studentNumber = None, semester = None):
        self.__studentNumber = studentNumber
        self.__year = year
        self.__semester = semester
        self.__studentList = []

    def getStudentNumber(self):
        return self.__studentNumber

    def setStudentNumber(self, studentNumber):
        self.__studentNumber = studentNumber

    def getSemester(self):
        return self.__semester

    def setSemester(self, semester):
        if semester < 1 or semester > 8:
            # If enter a value that less than 1 or more than 8 print error message
            raise ValueError("Semester can't be both less than 1 and more than 8.")
        self.__semester = semester


    def getYear(self):
        return self.__year

    def setYear(self, year):
        if year < 1 or year > 4:
            # If enter a value that less than 1 or more than 4 print error message
            raise ValueError("Year can't be both less than 1 and more than 4.")
        self.__year = year

    def __addStudentList(self, student):
        self.getStudentList().append(student)

    def getStudentList(self):
        return self.__studentList

    def setStudentList(self, studentList):
        self.__studentList = studentList
